### PR TITLE
[WIP] Hapi 17 Support

### LINKED ISSUE #330: hapi v17 support

### PR DESCRIPTION
**Pull request by ChristianMurphy**
TODO:
- [ ] leverage `async` and `await`
- [ ] leverage destructured assignment
- [ ] Update linter to allow async arrow functions
- [ ] Identify remaining gaps

resolves #330

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,15 @@
 language: node_js
 
 node_js:
-    - "4"
-    - "6"
```
Comment (Marsup) [inline .travis.yml]: Also add "9".

File: API.md [hunk 2]
```diff
@@ -14,9 +14,7 @@ combine **bell** with the [**hapi-auth-cookie**](https://github.com/hapijs/hapi-
 
 ```javascript
 var Hapi = require('hapi');
-var server = new Hapi.Server();
-
-server.connection({ port: 8000 });
+var server = new Hapi.Server({ port: 8000 });
```
Comment (Marsup) [inline API.md]: Hapi 17 doesn't need a `new`.

File: package.json [hunk 3]
```diff
@@ -38,28 +38,28 @@
...
+    "hapi": "17.x.x",
+    "hawk": "7.x.x",
+    "lab": "15.x.x",
+    "code": "5.x.x",
+    "sinon": "3.x.x"
   },
   "scripts": {
-    "test": "node node_modules/lab/bin/lab -t 100 -L -v",
-    "test-cov-html": "node node_modules/lab/bin/lab -r html -o coverage.html -L"
+    "test": "lab -t 100 -L -v",
```
Comment (Marsup) [inline package.json]: Should probably add back `-a code` to both since lab is back to no default assertion library.

File: API.md (hunk 2 above)
Comment (ChristianMurphy) [reply API.md]: Updated

File: .travis.yml (hunk 1 above)
Comment (ChristianMurphy) [reply .travis.yml]: Updated

File: package.json (hunk 3 above)
Comment (ChristianMurphy) [reply package.json]: added

File: API.md [hunk 4]
```diff
@@ -13,10 +13,8 @@ that require loading a single resource. Once the handler is called, the applicat
 combine **bell** with the [**hapi-auth-cookie**](https://github.com/hapijs/hapi-auth-cookie) authentication scheme plugin.
 
 ```javascript
-var Hapi = require('hapi');
-var server = new Hapi.Server();
-
-server.connection({ port: 8000 });
+var { Server } = require('hapi');
```
Comment (Marsup) [inline API.md]: Might as well use `const`s and other ES6 features wherever possible now.

Comment (ChristianMurphy) [reply API.md]: Updated

**Comment by ldesplat** [comment]
I mistakenly pushed to your branch but I forced it back to what it was :)

This hapi 17 upgrade is very non trivial for bell and all its tests.

**Comment by ChristianMurphy** [comment]
Closing, in favor of #334

**Pull request closed without merging**